### PR TITLE
[improvement](nereids) Support USE statement in nereids optimizer

### PR DESCRIPTION
**Pull request by liutang123**
## Proposed changes
Support USE statement in nereids optimizer
Issue Number: close #xxx

### CODE REVIEW

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -51,9 +51,14 @@ statementBase
     | materailizedViewStatement         #materailizedViewStatementAlias
     | constraintStatement               #constraintStatementAlias
     | supportedDropStatement            #supportedDropStatementAlias
+    | useStatement                      #useStatementAlias
     | unsupportedStatement              #unsupported
     ;
 
+useStatement
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:58]:
```suggestion
supportedUseStatement
```

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 2]
```diff
@@ -314,8 +319,7 @@ unsupoortedUnsetStatement
     ;
 
 unsupportedUseStatement
-    : USE (catalog=identifier DOT)? database=identifier                              #useDatabase
-    | USE ((catalog=identifier DOT)? database=identifier)? ATSIGN cluster=identifier #useCloudCluster
+    : USE ((catalog=identifier DOT)? database=identifier)? ATSIGN cluster=identifier #useCloudCluster
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:322]: could u help support `use db@cluster` too?

**Pull request closed without merging**